packages/cfd/src/Components/__tests__/cfd-mt5-demo-account-display.spec.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import CFDMT5DemoAccountDisplay from '../cfd-mt5-demo-account-display';
+import { CFD_PLATFORMS } from '@deriv/shared';
+
+//mock cfd-account-card
+jest.mock('../cfd-account-card', () => ({
+    CFDAccountCard: () => <div>CFDAccountCard</div>,
+}));
+
+describe('CFDMT5DemoAccountDisplay', () => {
+    const mock_standpoint = {
+        financial_company: 'financial_company',
+        gaming_company: 'gaming_company',
+        iom: false,
+        malta: false,
+        maltainvest: false,
+        svg: true,
+    };
+
+    const mock_props: React.ComponentProps<typeof CFDMT5DemoAccountDisplay> = {
+        is_eu: false,
+        is_eu_country: false,
+        has_maltainvest_account: false,
+        has_cfd_account_error: false,
+        openAccountNeededModal: () => jest.fn(),
+        standpoint: mock_standpoint,
+        is_loading: false,
+        is_logged_in: true,
+        isSyntheticCardVisible: jest.fn().mockReturnValue(true),
+        isFinancialCardVisible: jest.fn().mockReturnValue(false),
+        onSelectAccount: jest.fn(),
+        openAccountTransfer: jest.fn(),
+        platform: CFD_PLATFORMS.MT5,
+        current_list: {},
+        openPasswordManager: jest.fn().mockReturnValue(true),
+        residence: 'country',
+        landing_companies: {},
+        toggleMT5TradeModal: jest.fn(),
+        show_eu_related_content: false,
+    };
+    it('should render a loading component when is_loading is true', () => {
+        render(<CFDMT5DemoAccountDisplay {...mock_props} is_loading={true} />);
+        expect(screen.getByTestId('dt_barspinner')).toBeInTheDocument();
+    });
+
+    it('should render a loading component when is_loading is false', () => {
+        render(<CFDMT5DemoAccountDisplay {...mock_props} />);
+
+        expect(screen.getByTestId('dt_cfd_demo_accounts_display')).toHaveClass('cfd-demo-accounts-display');
+        expect(screen.getByTestId('dt_cfd_demo_accounts_display')).toBeInTheDocument();
+    });
+
+    it('should render the synthetic account card when isSyntheticCardVisible returns true', () => {
+        mock_props.current_list = { 'mt5.demo.synthetic.01': {} };
```

Comment:
Just asking, please ignore, if it's not relevant. What is the purpose of adding `mt5.demo.synthetic.01` here, why not in the very beginning, where you define `const mock_props`? Because `isSyntheticCardVisible` always returns `true` (line 30)

Reply:
And don't we need to check whether `isFinancialCardVisible` is `true` or `false`, because it affects on the rendering of the child component?

Reply:
The thing is the isFinancialVisible is dependent on the child component which is using connect so have to wait for the useStore card to merged after that need to update this test case.
Meanwhile the child component has already its test cases so no need to check those for now.
cc: @kate-deriv

Reply:
Good point👍🏻
And what about line 55? ([link](https://github.com/binary-com/deriv-app/pull/8477/files#r1185856428))